app/jobs/find_available_user_barcode_job.rb

```diff
@@ -38,8 +38,13 @@ def perform(user: nil, pin_code: nil, **args)
         # so that the ActiveJob mechanism would take care of retrying.
         # However, such exception handling will become available to us once
         # we upgrade our rails version.  For now, go simpler.
-        Delayed::Worker.logger.error("#{self.class.name}: user.check_barcode_uniqueness_with_sierra threw an error: #{exception.message || 'nil'}")
+        Delayed::Worker.logger.error("#{self.class.name}: user.check_barcode_uniqueness_with_sierra threw a communications error: #{exception.message || 'nil'}")
         raise exception
+      rescue ArgumentError => exception
+        # this is extremely unlikely, but shouldn't be insurmountable.
+        # let's log, and ask the following code to try again.
```

Comment:
@daryanypl In failure case why don't we send email to us. Since it is very important functionality.

Reply:
Good question.  I'm relying on the code block below to retry the functionality.  This is not the best exception handling.  When we go to a higher version of rails, we'll be able to do exception handling as it's meant to be done -- there will be hooks to catch exceptions, hooks to retry_on, a mechanism for controlling how many retries and at what time intervals from the config, and a hook for what to do when all retries fail.  What you see right now is rather clumsy, and is there holding space until we can make it better.

Reply:
Let me know if you still want an email sent.  I'm ok doing that, just not sure I should.

Reply:
Other than logs we dont have any alarm for this functionality. Better to send email

Reply:
@daryanypl  This Pr related to test cases, Please add email notification code here.